build-system/tasks/karma.conf.js

```diff
@@ -69,6 +78,19 @@ module.exports = {
     transform: [['babelify', {caller: {name: 'test'}, global: true}]],
     // Prevent "cannot find module" errors on Travis. See #14166.
     bundleDelay: isTravisBuild() ? 5000 : 1200,
+
+    persistentCache: browserifyPersistFs(
+      './build/.karma-cache',
+      {
+        deps: createHash(fs.readFileSync('./yarn.lock')),
+        build: globby.sync('build-system/**/*.js').map((f) => {
+          return createHash(fs.readFileSync(f));
+        }),
```

Comment:
What's meant to be captured here is "can browserify produce a different output". That means any changes to babel-plugins, or changes to a browserify setting, etc, could cause a different output. We could restrict this to just caching babel-plugin folders, but I didn't know where browserify settings could be changed so opt-ed for everything.

Reply:
> how long does it take to compute this hash?

I reduced the inputs. It takes roughly ~20ms for my 2015 MBP.

Reply:
Thanks for clarifying. The concern I had (and still have) is that if you do not include the test files (contained in `preprocessors`) in the input to `createHash`, the following workflow will break:

1. Check out `master`
2. Run `gulp unit --files test/unit/test-foo.js` (warms up the `browserify` cache)
3. Edit `test/unit/test-foo.js`
4. Run `gulp unit --files test/unit/test-foo.js` again

In step 4, since none of the contents of `build-system/` have changed, won't Karma used the previously cached transform of `test/unit/test-foo.js` instead of re-transforming it?

That's why I suggested that in order to have faith in the validity of the cache, it should change whenever _any_ of the files in `preprocessors` changes.

Or am I completely misunderstanding the purpose of this cache?

Reply:
Tested this out now that you've merged the PR, and realized this isn't a problem. Modifying a test file will cause it to be re-transformed. 👍